### PR TITLE
docs: Add more documentation around instrumentation libraries.

### PR DESCRIPTION
**Pull request by fhwang**
Not sure if this is the final state of this but it seemed like a good start. After thinking about it I opted against listing all the instrumentation libraries in any READMEs: This seems brittle and I'd imagine most users should be able to navigate Github and/or the repository on their own with a quick pointer. But I'm definitely open to other thoughts if we think there are ways this documentation can be clearer.

### CODE REVIEW

**Comment by fbogsany** [comment]
Part of the goal here is to have a table of what we currently support, what we're planning to work on, and what we know people might want but no-one is planning to work on yet. People won't necessarily know how to navigate the repo, and they shouldn't need to unless they plan to contribute.

A way to address "what we currently support" is #304. That doesn't address what we plan and don't plan to work on.

We could make sure we have issues for all the instrumentation we think people want, with ones we plan to implement assigned to the people who intend to implement them and all others marked with `#help-wanted`. Then we can add prominent links to the appropriate issue queries in the README.

File: instrumentation/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## License
+
+All OpenTelemetry Ruby instrumentation gems are distributed under the Apache 2.0 license. See [LICENSE][license-github] for more information.
+
+
+[license-github]: https://github.com/open-telemetry/opentelemetry-ruby/blob/main/LICENSE
+[ruby-sig]: https://github.com/open-telemetry/community#ruby-sig
+[community-meetings]: https://github.com/open-telemetry/community#community-meetings
+[ruby-gitter]: https://gitter.im/open-telemetry/opentelemetry-ruby
```
Comment (fbogsany) [inline instrumentation/README.md:50]: Gitter is going away: #620

**Comment by fbogsany** [comment]
"Instrumentation wanted" link: https://github.com/open-telemetry/opentelemetry-ruby/labels/instrumentation

"Instrumentation planned" link: https://github.com/open-telemetry/opentelemetry-ruby/issues?q=is%3Aopen+label%3Ainstrumentation+-label%3A%22help+wanted%22+

"Instrumentation help wanted" link: https://github.com/open-telemetry/opentelemetry-ruby/issues?q=is%3Aopen+label%3Ainstrumentation+label%3A%22help+wanted%22

**Comment by fhwang** [comment]
Thanks for the feedback @fbogsany . Just to be clear, do you believe this README should contain a table of what we currently support, in addition to a link to the OTel Registry? Or just the link? I'm inclined to think that the link will be easy enough and will always be up-to-date.

And I'm happy to take on #304 as well. Seems like we ought to be able to generate those registry PRs automatically.

**Comment by fbogsany** [comment]
Just the link is fine - we should only have to update one place for available instrumentation: the registry. Likewise for the "instrumentation we're going to implement" and "instrumentation people might want that we have no plans for yet".

**Comment by fhwang** [comment]
Closes #609.

**Review by fbogsany — APPROVED** [review]
This looks great, thanks @fhwang !

**Pull request merged**